test/fixtures/message/assert_throws_stack.snapshot

```diff
@@ -19,16 +19,16 @@ AssertionError [ERR_ASSERTION]: Expected values to be strictly deep-equal:
   generatedMessage: true,
   code: 'ERR_ASSERTION',
   actual: Error: foo
-      at assert.throws.bar (*assert_throws_stack.js:*)
-      at getActual (node:assert:*)
-      at Function.throws (node:assert:*)
-      at Object.<anonymous> (*assert_throws_stack.js:*:*)
```

Comment:
This test is called `assert_throws_stack`; I would think that this explicit stack trace is intended as part of the test?

As a broader question, how do we handle cases like (maybe) this one where the test won’t be using the `.snapshot` file as generated? Just edit them by hand after generation and hope that no one regenerates them later and overwrites? Use `spawnPromisified` instead?

Reply:
I am not sure I understood the question. editing the snapshot file will cause the test to fail

Reply:
> I am not sure I understood the question. editing the snapshot file will cause the test to fail

To use this case as an example, I assume the test/snapshot author wanted to test that the first line of the stack trace includes `at assert.throws.bar`. The generated snapshot just has `*` there, so the author would presumably replace `*` with something like the old `.out` file’s `at assert.throws.bar (*assert_throws_stack.js:*)`. I would think that this shouldn’t cause the test to fail—we’re replacing a very loose match with a specific match, but the specific match still passes against the input. However the process by which we’ve created this more specific snapshot file involved manual editing; is this the process you would recommend for tests like this where part of what we want to assert isn’t getting output in the generated snapshot?

Reply:
the snapshotting algorithm does not use regex or anything else to perform matching. matching is strict and all the replacements are just done before the comparison, so in this case we should change the string transforming to something less extensive

Reply:
So what’s the answer to my question? If the test author desires a test more specific than the one the snapshot would provide, are we expecting the test author to manually change the snapshot after generation?

Reply:
the test author would have to write some code transforming the output to ignore parts that are not important and keep parts that are, for example:
```js
const lines = output.split('\n');
const important = lines.slice(0,5).join('\n');
const ignored = snapshot.replaceStackTrace(lines.slice(5).join('\n'));
return `${important}${ignored}`
```
at this point of time my goal is to have the snapshots look pretty much the same as the current message tests,
but over time - if we see common patterns we can obviously make these transformations common